cpp/log/cert.cc

```diff
@@ -966,15 +971,20 @@ util::Status Cert::IsValidNameConstrainedIntermediateCa() const {
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
   StatusOr<bool> has_ca_constraint = HasBasicConstraintCATrue();
+  StatusOr<bool> has_name_constraints = HasExtension(NID_name_constraints);
+
   if (!has_ca_constraint.ok() || !has_ca_constraint.ValueOrDie()
-      || HasExtension(NID_name_constraints) == Cert::FALSE) {
+      || !has_name_constraints.ok() || !has_name_constraints.ValueOrDie()) {
```

Comment:
This is different - previously failing to get the `NID_name_constraints` extension would not have returned `OK`, but an `INVALID_ARGUMENT` below.

Arguably that's misleading, but perhaps we could do something like
```C++
if (!has_ca_constraint.ok()) {
  return has_ca_constraint.status();
}
if (!has_name_constraints.ok()) {
  return has_name_constraints.status();
}
if (!has_ca_constraint.ValueOrDie() || !has_name_constraints.ValueOrDie()) {
  return OK;
}
```